x-pack/plugins/session_view/public/components/process_tree/helpers.ts

```diff
@@ -11,9 +11,43 @@ import {
   Process,
   ProcessEvent,
   ProcessMap,
+  ProcessFields,
 } from '../../../common/types/process_tree';
 import { ProcessImpl } from './hooks';
 
+// Creates an instance of Process, from a nested leader process fieldset
+// This is used to ensure we always have a record for a session leader, as well as
+// a parent record for potentially orphaned processes
+export function inferProcessFromLeaderInfo(sourceEvent?: ProcessEvent, leader?: ProcessFields) {
+  const entityId = leader?.entity_id || '';
```

Comment:
are we using this `entityId` for React keys? if so would be nice to avoid duplicated keys i.e `const entityId = leader?.entity_id || uuid();`